### PR TITLE
CFIN-471 Allow different Funnelback profiles to be queried per environment

### PR DESCRIPTION
**Pull request by ce666-york**
This PR changes the API to query the `_default` Funnelback profile in the live environment, and the `default_preview` profile in development.

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -0,0 +1,5 @@
+BASE_URL=https://www.york.ac.uk/search/
+COLLECTION=york-uni-courses
+FORM=course-search
+PROFILE=_default_preview
+SMETA_CONTENT_TYPE=course
```
Comment (ce666-york) [inline .env.development:5]: After discussing with Richard we decided to move all of the constants to config files, even the ones that are the same across environments - this is the same as Course Search.

**Review by FergusMcGlynn — APPROVED** [review]

File: src/utils/constructFunnelbackUrls.test.js [hunk 2]
```diff
@@ -1,7 +1,12 @@
 const { coursesUrl } = require("./constructFunnelbackUrls");
-const { BASE_URL, COLLECTION, FORM, PROFILE, SMETA_CONTENT_TYPE } = require("../constants/UrlAndParameters");
 
-const constantPartOfSearchUrl = `${BASE_URL}?collection=${COLLECTION}&form=${FORM}&profile=${PROFILE}&smeta_contentType=${SMETA_CONTENT_TYPE}`;
+const constantPartOfSearchUrl = `${process.env.BASE_URL}?collection=${process.env.COLLECTION}&form=${process.env.FORM}&profile=${process.env.PROFILE}&smeta_contentType=${process.env.SMETA_CONTENT_TYPE}`;
+
+test("constructs the Funnelback url with the expected environment variables", async () => {
```
Comment (RichardAllum) [inline src/utils/constructFunnelbackUrls.test.js:5]: Does this test need to be `async`?

**Review by RichardAllum — APPROVED** [review]

File: src/utils/constructFunnelbackUrls.test.js (hunk 2 above)
Comment (ce666-york) [reply src/utils/constructFunnelbackUrls.test.js]: Courtesy of our good friends "copy" and "paste". Fixed now

File: package.json [hunk 3]
```diff
@@ -8,20 +8,24 @@
...
     "formatandcheck": "npm run format && npm run lint && npm run test",
+    "fc": "npm run formatandcheck",
     "check": "npm run checkformat && npm run lint && npm run test",
-    "package": "serverless package",
-    "deploy": "serverless deploy",
+    "package": "serverless package --env production",
+    "deploy": "serverless deploy --env production",
+    "deploy:dev": "serverless deploy --env development",
     "undeploy": "serverless remove"
   },
```
Comment (ncn1) [inline package.json:17]: I know this has been merged, but should the undeploy command also specify environment?

Comment (ce666-york) [reply package.json:17]: It's a good question - I don't think it's necessary, I believe the Serverless commands will run against whatever AWS account you're logged in as via `saml2aws` - the `--env` flag just tells Serverless what `.env` file to use

**Pull request merged**